neuraxle/base.py

```diff
@@ -227,10 +234,12 @@ class ExecutionContext:
 
     def __init__(
             self,
+            execution_mode: ExecutionMode,
```

Comment:
Note that even in fit mode, some steps will fit_transform when they need to pass data to a step after themselves in a pipeline.